src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/HighlightUtils.java

```diff
@@ -128,19 +171,27 @@ private static HighlightSettings createGlobalFieldSettings(
         settings.hasHighlightQuery()
             ? QUERY_NODE_MAPPER.getQuery(settings.getHighlightQuery(), indexState)
             : searchQuery;
-    builder.withFieldQuery(getFieldQuery(indexReader, query, settings.getFieldMatch()));
+    builder.withHighlightQuery(query);
 
     return builder.build();
   }
 
-  private static FieldQuery getFieldQuery(
-      IndexReader indexReader, IndexState indexState, Settings settings) throws IOException {
-    Query query = QUERY_NODE_MAPPER.getQuery(settings.getHighlightQuery(), indexState);
-    return getFieldQuery(indexReader, query, settings.getFieldMatch());
-  }
-
-  private static FieldQuery getFieldQuery(IndexReader indexReader, Query query, boolean fieldMatch)
-      throws IOException {
-    return new FieldQuery(query, indexReader, true, fieldMatch);
+  private static String resolveHighlighterName(Settings settings) {
+    switch (settings.getHighlighterType()) {
+      case DEFAULT:
+        /* default -> default-highlighter is only applicable in global settings.
+         * In field override, we should return the one in global settings instead.
+         */
+        return DEFAULT_HIGHLIGHTER_NAME;
+      case PLAIN:
+        throw new UnsupportedOperationException("plain-highlighter is not supported yet.");
+      case FAST_VECTOR:
+        return NRTFastVectorHighlighter.HIGHLIGHTER_NAME;
+      case CUSTOM:
+        return settings.getCustomHighlighterName();
+      default:
+        throw new IllegalArgumentException(
+            String.format("Unknown highlighter_type [%s]", settings.getHighlighterType()));
```

Comment:
Nit: change to `"Unknown highlighter_type: %s"` so it's formatted similar to most other errors. You can also use just `+` to append the strings if you want